src/services/BookingService.ts

```diff
@@ -0,0 +1,97 @@
+import { Brackets } from "typeorm";
+import bookingRepository from "../repositories/BookingRepository";
+import IRepository from "../types/IRepository";
+import Booking from "../entities/Booking";
+import BookingCreateBodyValidator from "../validators/Booking/BookingCreateBodyValidator";
+import SlotService from "./SlotService";
+import HttpException from "../exceptions/HttpException";
+
+
+class BookingService {
+
+  public getRepository(): IRepository<Booking> {
+    return bookingRepository();
+  }
+
+  public async isSlotAvalible(
+    slotId: number,
+    bookingData: BookingCreateBodyValidator,
```

Comment:
Let's be more explicit. Remember _interface segregation_ principle. Avoid coupling to an interface you don't need, that means **be explicit** to what you need. Do all clients that want to use this method need to support `BookingCreateBodyValidator` even if the interface keeps being extended? Probably not. You only need `slotId`, `startDate` and `endDate` so far, so let that be the signature of this method and not a `BookingCreateBodyValidator` interface.